DataFormats/L1TrackTrigger/interface/TTTrack.h

```diff
@@ -201,6 +207,7 @@ TTTrack<T>::TTTrack() {
   thePhiSector_ = 0;
   theEtaSector_ = 0;
   theTrackSeedType_ = 0;
+  theGTTLinkIndex_ = -1;
```

Comment:
Couple of comments here (provided this isn't some direct firmware port):

1. I see that this GTT link index variable is getting added here, and adding a few utilities for retrieving it and reading it, but it is default initialized to (what I assume is) an invalid value. Would it make sense for the places that use this later with `gttLinkIndex()` to check validity of the link index (i.e. not -1?)
2. This isn't necessarily something for this PR, but I think the context of this line is a class constructor for `TTTrack`, so I think CMSSW preference on this kind of assignment is the member assignment/initialization list

Reply:
1. Yes, this is explicitly checked in the free selector function (https://github.com/cms-sw/cmssw/pull/42505/files#diff-8a1bfc099e89c89bc55c2dea9126fccd72614c3e3df753a358f013f6e8f813c5R103) . Alternatively, it could be made an unsigned int and set very high. 
2. Can you expand a bit? But just to be explicit, this variable isn't 'known' until the track is (virtually) handed off to the GTT subsystem, so it's not intrinsic to the track itself, but only capable of being computed 'at the subsystem boundary' as it were, a second reason the member variable is initialized to an invalid value.